modules/aggregations/src/test/java/org/elasticsearch/aggregations/bucket/histogram/InternalAutoDateHistogramTests.java

```diff
@@ -65,12 +65,11 @@ protected SearchPlugin registerPlugin() {
         return new AggregationsPlugin();
     }
 
-    // TODO: the base test class should be able to get this from the search plugin? (^)
```

Comment:
Here we're registering the high level REST client's version which isn't in the SearchPlugin.

Reply:
I wonder if we should be removing these as we go. We don't need to parse the thing any more.

Reply:
I thought we also had these tests to test the xcontent serialization (similarly how we test the binary serialization)? We lose that if we drop these tests?

Reply:
(the downside is that we need to maintain the xcontent parsing in tests...)

Reply:
Yeah. That's why I didn't nuke it. It's not obviously right. Before we had the high level REST client we never tested the round trip here - we'd just render xcontent and test in REST tests or specific xcontent renderings. Now we have randomized parsing. But it's quite heavy.